client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -198,7 +199,7 @@ private Thread startDriver(final RpcServer rpcServer, final String clientId, fin
     Runnable runnable;
     final String serverAddress = rpcServer.getAddress();
     final String serverPort = String.valueOf(rpcServer.getPort());
-
+    final SparkLauncher launcher = new SparkLauncher();
```

Comment:
This is only used inside the `else` in L230, so it's better to declare it there.